### PR TITLE
Update UG with find function, update various DG sections

### PR DESCRIPTION
**Pull request by LMAOboxhack**
Added `find` function to Features section in User Guide.
Added target user profile, value proposition, non-functional requirements, user story & use cases for `find` function to Developer Guide.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -274,12 +280,18 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 | -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
 | `* * *`  | general user                               | add a new applicant under a position      | store his information within the system                                |
 | `* * *`  | general user                               | delete an applicant from under a position | Remove applicants that are no longer related to this position          |
+| `* * *`  | new user                                   | see usage instructions         | refer to instructions when I forget how to use the App                 |
+| `* * *`  | user                                       | add a new person               |                                                                        |
+| `* * *`  | user                                       | delete a person                | remove entries that I no longer need                                   |
+| `* * *`  | user                                       | easily search all relevant applicants through keywords         | I can compare applicants' relevant information |
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:286]:
```suggestion
| `* * *`  | user                                       | easily search all relevant applicants through keywords         | compare applicants' relevant information |
```
'I can' is already at the header of the table.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -274,12 +280,18 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
+| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:288]:
```suggestion
| `*`      | user with many applicants in the address book | sort applicants by name           | locate an applicant easily                                                 |
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -319,14 +331,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
     * 3a1. MrTechRecruiter shows an error message.
 
       Use case resumes at step 2.
-    
+
+**Use case: Search for applicants through keywords**
+
+**MSS**
+
+1. User enters the keywords to search by
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:339]:
```suggestion
1. User enters the keywords for MrTechRecruiter to search for.
```
All use case statements need full stops/periods at the end. Comment applied for all subsequent sentences.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -319,14 +331,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. MrTechRecruiter performs the search according to the keywords
+3. MrTechRecruiter displays the search results
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:349]: The line 'use case ends' needs to be same indentation as the start of the sentence.
```suggestion
        Use case ends.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -319,14 +331,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+
+* 1a. An invalid flag is specified.
+
+    * 1a1. AddressBook shows an error message.
+
+      Use case ends.
```
Comment (BenjaminTan99) [inline docs/DeveloperGuide.md:356]: Same situation here. Also extension 1a should come before 2a.

**Review by BenjaminTan99 — CHANGES_REQUESTED** [review]
Some edits required before pushing.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -319,14 +331,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4. Should be portable and work without requiring an installer.
+5. Should not depend on a remote server.
+6. Should work for standard screen resolutions and higher.
```
Comment (Harry-Gao-H) [inline docs/DeveloperGuide.md:367]: Maybe the `portable` here could be ambiguous.
And `standard screen resolutions` can be specified to `industry-standard Full HD 1080p resolution`?

**Review by Harry-Gao-H — COMMENTED** [review]
Some minor things to change, otherwise LGTM

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (LMAOboxhack) [reply docs/DeveloperGuide.md:339]: Will fix, thanks!

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (LMAOboxhack) [reply docs/DeveloperGuide.md:356]: Will fix, thanks!

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -319,14 +331,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+
+* 1a. An invalid flag is specified.
+
+    * 1a1. AddressBook shows an error message.
```
[the remark this answers is not in the document]
Comment (LMAOboxhack) [reply docs/DeveloperGuide.md:354]: Will fix, thanks!

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (LMAOboxhack) [reply docs/DeveloperGuide.md:367]: Will fix, thanks!

**Review by BenjaminTan99 — APPROVED** [review]

**Pull request merged**